### PR TITLE
Remove the Unicode BOM, if present. (#97)

### PR DESCRIPTION
**Pull request by lardieri**
CSV files saved by Excel and Airtable include the BOM when encoding to UTF-8.

### CODE REVIEW

File: SwiftCSV/CSV.swift [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 import Foundation
 
+fileprivate let bom = "\u{FEFF}"
```
Comment (DivineDominion) [inline SwiftCSV/CSV.swift:11]: I'd like to suggest two changes here:

1) Spell out `byteOrderMark` to make obvious what "bom" is supposed to mean and help other devs google the term;
2) Add a doc comment with a link to https://github.com/swiftcsv/SwiftCSV/issues/97, and/or copy parts your own super useful explanation (thanks for that!) from there.

File: SwiftCSVTests/utf8_with_bom.csv [hunk 2]
```diff
@@ -0,0 +1,2 @@
+﻿Part Number,Description,Unit Price,Qty
+12345,Heizölrückstoßabdämpfung,"€ 100,00",2
```
Comment (DivineDominion) [inline SwiftCSVTests/utf8_with_bom.csv:2]: :) Thanks for that example -- and the test

**Review by DivineDominion — CHANGES_REQUESTED** [review]
LGTM, but I personally would prefer a change to the "`bom`" constant to make its meaning more obvious.

Unrelated to that, WDYT about merging your 0.6.2 fixes with this repo, too, so we can release that to CocoaPods proper>?

**Comment by lardieri** [comment]
I've added additional documentation around the BOM.

I'm happy to submit 0.6.2 for release, but you will need to create a branch in your repo for me to open PR against, since it forks from a previous point (039cc273).

**Review by DivineDominion — APPROVED** [review]
LGTM 👍 Thanks for the doc comment!

**Comment by DivineDominion** [comment]
I've opened the branch `v0.6`. On top of the v0.6.1 release, it includes delimiter guessing, which was compatible with the v0.6 code. Will that be causing any trouble? Happy to discuss in another PR! :)

**Comment by DivineDominion** [comment]
Also, thanks a lot for contributing to SwiftCSV! We've invited you to join the SwiftCSV GitHub organization – no pressure to accept! If you'd like more information on what that means, check out our [contributor guidelines](https://github.com/SwiftCSV/SwiftCSV/blob/master/CONTRIBUTING.md).

**Pull request merged**